src/sentry/static/sentry/app/views/settings/components/forms/field/index.tsx

```diff
@@ -16,135 +15,92 @@ import FieldErrorReason from 'app/views/settings/components/forms/field/fieldErr
+  InheritedFieldWrapperProps &
+  InheritedControlStateProps & {
     /**
      * Show "required" indicator
      */
-    required: PropTypes.bool,
-
+    required?: boolean;
     /**
      * Should field be visible
      */
-    visible: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
-
+    visible?: boolean | ((props: Props) => boolean);
     /**
      * Should field be disabled?
      */
-    disabled: PropTypes.oneOfType([PropTypes.bool, PropTypes.func]),
-
-    /**
-     * Reason why field is disabled (displays in tooltip)
-     */
-    disabledReason: PropTypes.string,
-
-    /**
-     * Displays the help element in the tooltip
-     */
-    showHelpInTooltip: PropTypes.bool,
-
-    /**
-     * Error message
-     */
-    error: PropTypes.string,
-
+    disabled?: boolean | ((props: Props) => boolean);
     /**
-     * Hide ControlState component
+     * User-facing field name
      */
-    flexibleControlStateSize: PropTypes.bool,
-
+    label?: React.ReactNode;
     /**
-     * User-facing field name
+     * Should the label be rendered for the field?
      */
-    label: PropTypes.node,
-
+    hideLabel?: boolean;
     /**
      * Help or description of the field
      */
-    help: PropTypes.oneOfType([PropTypes.node, PropTypes.element, PropTypes.func]),
-
+    help?: React.ReactNode | React.ReactElement | ((props: Props) => React.ReactNode);
     /**
-     * Should Control be inline with Label
+     * Displays the help element in the tooltip
      */
-    inline: PropTypes.bool,
-
+    showHelpInTooltip?: boolean;
     /**
-     * Should the field display in a stacked manner (no borders + reduced padding
+     * The control's `id` property
      */
-    stacked: PropTypes.bool,
-
+    id?: string;
     /**
-     * The control's `id` property
+     * Additional inline styles for the field
      */
-    id: PropTypes.string,
-
+    style?: React.CSSProperties;
     /**
-     * Field is in saving state
+     * The classname of the field
      */
-    isSaving: PropTypes.bool,
-
+    className?: string;
     /**
-     * Field has finished saving state
+     * The classname of the field control
      */
-    isSaved: PropTypes.bool,
-
+    controlClassName?: string;
     /**
-     * The Control component
+     * Error message to display for the field
      */
-    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
-
+    error?: string;
+    validate?: Function; //TODO(TS): Do we need this?
     /**
-     * Class name for inner control
+     * The control to render. May be given a function to render with resolved
+     * props.
      */
-    controlClassName: PropTypes.string,
-
-    /** Inline style */
-    style: PropTypes.object,
+    children?: React.ReactNode | ((props: ChildRenderProps) => React.ReactNode);
   };
 
+type ChildRenderProps = Omit<Props, 'className' | 'disabled'> & {
+  help: React.ReactNode;
+  disabled?: boolean;
+  errorState: React.ReactNode | null;
+  controlState: React.ReactNode;
```

Comment:
```suggestion
  help: React.ReactNode;
  errorState: React.ReactNode | null;
  controlState: React.ReactNode;
  disabled?: boolean;
```

Reply:
@EvanPurkhiser can please just move the optional prop to the end of the object? :)